### PR TITLE
fix(lerna): skip bootstrap if using a v7 range

### PR DESCRIPTION
**Pull request by rarkins**
## Context

Fixes #23482

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/npm/post-update/lerna.spec.ts [hunk 1]
```diff
@@ -184,33 +184,23 @@ describe('modules/manager/npm/post-update/lerna', () => {
       const res = await lernaHelper.generateLockFiles(
         lernaPkgFile('npm'),
         'some-dir',
-        { ...config, constraints: { ...config.constraints, npm: '6.0.0' } },
+        {
+          ...config,
+          constraints: { ...config.constraints, lerna: '^7.1.4', npm: '6.0.0' },
+        },
```
Comment (rarkins) [inline lib/modules/manager/npm/post-update/lerna.spec.ts:190]: I changed this test in order to trigger the "v7 range" code logic

File: lib/modules/manager/npm/post-update/lerna.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ import type { GenerateLockFileResult } from './types';
 import { getPackageManagerVersion, lazyLoadPackageJson } from './utils';
 
 // Exported for testability
-export function getLernaVersion(
+export function getLernaConstraint(
```
Comment (rarkins) [inline lib/modules/manager/npm/post-update/lerna.ts:25]: Renamed for correctness

File: lib/modules/manager/npm/post-update/lerna.ts [hunk 3]
```diff
@@ -115,18 +115,21 @@ export async function generateLockFiles(
       extraEnv.NPM_AUTH = env.NPM_AUTH;
       extraEnv.NPM_EMAIL = env.NPM_EMAIL;
     }
-    const lernaVersion =
+    const lernaConstraint =
```
Comment (rarkins) [inline lib/modules/manager/npm/post-update/lerna.ts:118]: Renamed for correctness

File: lib/modules/manager/npm/post-update/lerna.ts [hunk 4]
```diff
@@ -115,18 +115,21 @@ export async function generateLockFiles(
...
-      getLernaVersion(lernaPackageFile, await lazyPgkJson.getValue());
+      getLernaConstraint(lernaPackageFile, await lazyPgkJson.getValue());
     if (
-      !is.string(lernaVersion) ||
-      (semver.valid(lernaVersion) && semver.gte(lernaVersion, '7.0.0'))
+      !is.string(lernaConstraint) ||
+      (semver.valid(lernaConstraint) && semver.gte(lernaConstraint, '7.0.0')) ||
+      (semver.validRange(lernaConstraint) &&
+        (semver.satisfies('7.0.0', lernaConstraint) ||
+          semver.satisfies('7.999.999', lernaConstraint)))
```
Comment (rarkins) [inline lib/modules/manager/npm/post-update/lerna.ts:126]: This logic is added. A little hacky but should cover pretty much everyone. If I had more time I'd duplicate the containerbase util logic and resolve it fully

File: lib/modules/manager/npm/post-update/lerna.ts [hunk 5]
```diff
@@ -115,18 +115,21 @@ export async function generateLockFiles(
...
     ) {
-      logger.debug('Skipping lerna bootstrap');
+      logger.debug('Skipping lerna bootstrap for lerna >= 7.0.0');
       cmd.push(`${lernaClient} install ${cmdOptions}`);
```
Comment (viceice) [inline lib/modules/manager/npm/post-update/lerna.ts:129]: shouldn't we skip lerna global install completely to save runtime?

Comment (rarkins) [reply lib/modules/manager/npm/post-update/lerna.ts:129]: This `lernaClient` is actually npm, yarn, etc. I think we do skip lerna install as-is - check the snapshots

**Review by viceice — APPROVED** [review]

**Pull request merged**